product_pack/models/product_pack_line.py

```diff
@@ -21,6 +21,7 @@ class ProductPackLine(models.Model):
     )
     price = fields.Float()
 
-    @api.onchange('component_id')
-    def onchange_component_id(self):
-        self.price = self.component_id.list_price
+    @api.onchange('quantity', 'product_id')
+    def _onchange_price(self):
+        for rec in self:
```

Comment:
En los `onchange` no hay que usar un bucle `for`, solo se ejecutarán cuando el usuario modifique algo por lo que siempre es un solo registro